src/passes/pass.cpp

```diff
@@ -896,7 +896,9 @@ void PassRunner::runPass(Pass* pass) {
     checker = std::unique_ptr<AfterEffectModuleChecker>(
       new AfterEffectModuleChecker(wasm));
   }
-  pass->run(this, wasm);
+  assert(!pass->getPassRunner());
+  pass->setPassRunner(this);
+  pass->run(wasm);
```

Comment:
How about resetting the runner to nullptr afterwards? Or is it the idea that once a pass has run it can't run again anyhow? That seems right to me, but I'm not 100% sure. Might be worth a comment here as to why we don't reset to null.

Reply:
Will add a comment.